### PR TITLE
chore: update prisma + deno + rust deps

### PR DESCRIPTION
**Pull request by Yohe-Am**
- Bump deno to 1.46.3
- Update prisma-engines to 5.20
- Update other rust deps.

Closes MET-669 and MET-622 and MET-680.

#### Migration notes

...

- [ ] The change comes with new or modified tests
- [ ] Hard-to-understand functions have explanatory comments
- [ ] End-user documentation is updated to reflect the change

### CODE REVIEW

**Comment by michael-0acf4** [comment]
I will leave [this](https://github.com/metatypedev/deno/commit/36fe604eca8559edace8f06bbe6c90dc2242df6b#diff-bc437a803584ad992ce7c203b7d01159767cfa5d27b418677a8e3db872b780c3R76) here for future reference. Awesome work!

**Review by zifeo — DISMISSED** [review]
Nice!

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -82,7 +82,7 @@ repos:
       - id: check-merge-conflict
       - id: end-of-file-fixer
         # exclude all generated files
-        exclude: (?x)( .ghjk/.* |.*deno.lock |.*\.snap$|typegate/src/typegraphs/.*\.json |website/docs/reference/ |libs/pyrt_wit_wire/pyrt)
+        exclude: (?x)( .ghjk/.* |.*deno.lock |.*\.snap$|typegate/src/typegraphs/.*\.json |website/docs/reference/ |libs/pyrt_wit_wire/pyrt | migration_lock.toml)
```
[the remark this answers is not in the document]
Comment (Yohe-Am) [reply .pre-commit-config.yaml:85]: They're auto-generated files that get generated again on test runs. Fixing the EOF isn't helpful.

File: src/metagen/src/fdk_rust/static/Cargo.toml [hunk 2]
```diff
@@ -7,7 +7,7 @@ path = "lib.rs"
 crate-type = ["cdylib", "rlib"]
 
 [dependencies]
-anyhow = "1.0.86"
-serde = { version = "1.0.203", features = ["derive"] }
-serde_json = "1.0.117"
-wit-bindgen = "0.26.0"
+anyhow = "1.0.89"
+serde = { version = "1.0.210", features = ["derive"] }
```
[the remark this answers is not in the document]
Comment (Yohe-Am) [reply src/metagen/src/fdk_rust/static/Cargo.toml:11]: This was the result of using the `cargo update` tool. Cargo version ranges are `^x.y.z` by default so it should be good.

**Review by zifeo — APPROVED** [review]

**Pull request merged**